### PR TITLE
[DOCS] Update documentation to reference `get_context`

### PR DESCRIPTION
**Pull request by cdkini**
Changes proposed in this pull request:
- Replace existing references to `DataContext` and `BaseDataContext` with `get_context`

### Definition of Done
- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

File: docs/deployment_patterns/how_to_use_great_expectations_in_deepnote.md [hunk 1]
```diff
@@ -7,15 +7,15 @@ import TechnicalTag from '@site/docs/term_tags/_tag.mdx';
 
 _This piece of documentation was authored by [Allan Campopiano](https://www.linkedin.com/in/allan-campopiano-703394120)._
 
-This guide will help you get started with Great Expectations on Deepnote. You will learn how to validate columns in a Pandas DataFrame, 
+This guide will help you get started with Great Expectations on Deepnote. You will learn how to validate columns in a Pandas DataFrame,
```
Comment (cdkini) [inline docs/deployment_patterns/how_to_use_great_expectations_in_deepnote.md:10]: My editor strips unnecessary whitespace so please be aware that some of the diffs are just this

File: great_expectations/cli/checkpoint_script_template.py [hunk 2]
```diff
@@ -19,9 +19,9 @@
 import sys
 
 from great_expectations.checkpoint.types.checkpoint_result import CheckpointResult
-from great_expectations.data_context import DataContext
+from great_expectations.util import get_context
 
-data_context = DataContext(context_root_dir="{1:s}")
+data_context = get_context(context_root_dir="{1:s}")
```
Comment (Kilo59) [inline great_expectations/cli/checkpoint_script_template.py]:
```suggestion
data_context: FileDataContext = get_context(context_root_dir="{1:s}")
```
Should we annotate this since we are alos annotating the `CheckpointResult`? 🤷

**Review by Kilo59 — APPROVED** [review]
Non-blocking question

File: great_expectations/cli/checkpoint_script_template.py (hunk 2 above)
Comment (cdkini) [reply great_expectations/cli/checkpoint_script_template.py]: sure! doing that now
